galasa-parent/dev.galasa.framework.api.webui/src/main/java/dev/galasa/framework/api/webui/WorklistQuery.java

```diff
@@ -0,0 +1,173 @@
+		} catch (Exception e) {
+			throw new ServletException("Error while updating the DSS", e);
+		}
+		
+		int statusCode = 200;
+		if (worklistUtils.newResourceCreated == true) {
+			statusCode = 201;
+		}
+		
+		returnWorklist(principalUsername, resp, statusCode);
+ 
+	}
+	
+	
+	@Override
+	protected void doDelete(HttpServletRequest req, HttpServletResponse resp) throws ServletException, IOException {
+		
+		String principalUsername = worklistUtils.getPrincipalUsername(req);
+		
+		Map<String, String> paramMap = getParameterMap(req);
+		
+		String runId = "";
+		if (paramMap.get("runId") != null && !paramMap.get("runId").equals("")) {
+			runId = paramMap.get("runId");
+		}
+		
+
+		try {
+			worklistUtils.removeRunIdFromDss(principalUsername, runId);
+		} catch (Exception e) {
+			throw new ServletException("Error while updating the DSS", e);
+		}
+		
+		returnWorklist(principalUsername, resp, 200);
+ 
+	}
+	
+	
+	protected void returnWorklist(String principalUsername, HttpServletResponse resp, int statusCode) throws ServletException {
+		
+		List<String> runIdsInWorklist = new ArrayList<String>();
+		
+		try {
+			 runIdsInWorklist = worklistUtils.getRunIdsFromDss(principalUsername);
+		} catch (DynamicStatusStoreException e) {
+			throw new ServletException("Error while retrieving Worklist property from the DSS", e);
+		}
+		
+
+		List<WorklistItem> runs = worklistUtils.getWorklistItems(principalUsername, runIdsInWorklist);
+		
+		
+		JsonObject worklist = new JsonObject();
+		JsonElement worklistItems = gson.toJsonTree(runs);
+		worklist.add("worklistItems", worklistItems);
+		
+		
+		String json = "";
+		 
+		if (!worklist.isJsonNull()) {
+			try {
+				json = gson.toJson(worklist);
+			} catch (Exception e) {
+				throw new ServletException("An error occurred whilst retrieving the Worklist", e);
+			}
+		 }
+		
+		 try {
+	         PrintWriter out = resp.getWriter();
+	         resp.setContentType( "Application/json");
+	         resp.addHeader("Access-Control-Allow-Origin", "*");
+	         resp.setStatus(statusCode);
+	         out.print(json);
+	         out.close();
+	      } catch (Exception e) {
+	         throw new ServletException("An error occurred whilst retrieving the Worklist", e);
+	      }
+		 
+		
+	}
+
+	
+	private Map<String, String> getParameterMap(HttpServletRequest request) {
+
+	      Map<String, String[]> ParameterMap = request.getParameterMap();
+	      Map<String, String> newParameterMap = new HashMap<>();
+	      for (String parameterName : ParameterMap.keySet()) {
+	         String[] values = ParameterMap.get(parameterName);
+	         if (values != null && values.length > 0) {
+	            newParameterMap.put(parameterName, values[0]);
+	         } else {
+	            newParameterMap.put(parameterName, null);
+	         }
+	      }
+	   
+	      return newParameterMap;
+	}
+	
+	
+	@Activate
+	public void activate() {
+	   this.worklistUtils = new WorklistUtils(this.framework);
```

Comment:
I don't understand the reasoning on why the methods in WorklistUtils needs to be in a different class?  Please explain

Reply:
I thought there may be a chance in future that some of the methods used could also be used for the Compare List if that is implemented (albeit the name of the class would be better as just ListUtils for example) - methods such as getPrincipalUsername, getRunIdsFromDss, getRunById. They would need some refactoring to take the 'worklist' property as a parameter so it could be swapped with 'comparelist' but that's why I had it in a different class. Please let me know what you think.

Reply:
I don't expect the worklist stuff to be used outside of this servlet as this servlet is all about worklists.
The complication with the principal user is only temporary until I get authentication working,  in which case you can guarentee that there will be a principal user.
so, right thinking, but I don't believe this code will be reused in this instance.
Please merge in the methods.